src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(health, name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+  static removeFromAlive(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name, hidden) {
+    super(health, name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!(animal instanceof Herbivore) || animal.hidden) {
```

Comment:
The condition should check if 'animal' is an instance of Herbivore and is not hidden. However, the current condition will not work as expected because it will return early if 'animal' is an instance of Herbivore due to the negation operator. Change the condition to: 'if (animal instanceof Herbivore && !animal.hidden)'.